language/vm/vm_runtime/src/txn_executor.rs

```diff
@@ -271,7 +271,11 @@ where
                         if module_id == *EVENT_MODULE
                             && function_name == EMIT_EVENT_NAME.as_ident_str()
                         {
-                            let msg = self.execution_stack.pop_as::<ByteArray>()?;
+                            let msg = self
+                                .execution_stack
+                                .pop()?
+                                .simple_serialize()
```

Comment:
I'm kind of surprised that this change was enough to prevent any client-related tests from breaking. There must be code somewhere in the clients (and storage layer possible) that tries to interpret the `msg` part of a `ContractEvent`?